lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -41,10 +42,38 @@ def query_api(
             # which will only be made if the request is not forbidden nor unauthorized.
             if expected_status not in [HTTPStatus.FORBIDDEN, HTTPStatus.UNAUTHORIZED]:
                 rsp.add(responses.POST, es_url, **kwargs)
+                if scroll:
+                    scroll_payload = get_scroll_payload(host=host, port=port)
+                    rsp.add(
+                        responses.POST,
+                        scroll_payload.es_uri,
+                        json=scroll_payload.es_response,
+                        status=200,
+                    )
             response = client.post(
                 f"{server_config.rest_uri}{pbench_uri}", headers=headers, json=payload,
             )
         assert response.status_code == expected_status
         return response
 
     return query_api
+
+
+def get_scroll_payload(host: str, port: str) -> NamedTuple:
+    """
+    returns the
+    """
+    scroll_payload = NamedTuple("Scroll", [("es_uri", str), ("es_response", dict)],)
+    es_scroll_response = {
+        "_scroll_id": "same_random_scroll_id==",
+        "took": 3,
+        "timed_out": "false",
+        "_shards": {"total": 5, "successful": 5, "skipped": 0, "failed": 0},
+        "hits": {
+            "total": {"value": 50, "relation": "eq"},
+            "max_score": "null",
+            "hits": [],
+        },
```

Comment:
This is not going to have any discernible effect -- you need to return at least one hit so that, when this is joined together with the original response, the result is different from the original.

(That is, you should have two nearly-identical tests, one with `scroll` set to `True` and one with it `False`, and they should have different "expected results".)

Reply:
Yeah that's true but its probably little complicated to implement since we are mocking the scroll endpoint request that always return the same response, and so it would never come out of loop from inside the scroll function (since there will always be a response back from the scrolling)

Reply:
But anyway that should be doable with the help of reposnes callbacks https://github.com/getsentry/responses